certificate-manager/src/main/java/io/strimzi/certs/CertAndKey.java

```diff
@@ -4,16 +4,25 @@
  */
 package io.strimzi.certs;
 
+import java.nio.charset.Charset;
 import java.util.Base64;
 
 public class CertAndKey {
 
     private final byte[] key;
     private final byte[] cert;
+    private final byte[] store;
+    private final String storePassword;
 
     public CertAndKey(byte[] key, byte[] cert) {
+        this(key, cert, null, null);
+    }
+
+    public CertAndKey(byte[] key, byte[] cert, byte[] store, String storePassword) {
```

Comment:
Is this a truststore or keystore? My expectation is keystore. But from discusseions I understood you are working on public certs only. We should clarify it in the names and probably also add some Javadoc.

Reply:
So, this class actually is used for containing CA cert and related key but the "store" contains just the CA cert, so it's actually a "truststore". At the same time, the class is used in the KafkaUser model for containing User key and cert and I expect to use the `store` field to contain a "keystore" for TLS client authentication. This is way I was generic using just `store`.

Reply:
So how do you expect this method to look like for the user certificates? You will need to distinguish whether you want a truststore or keystore from this. If it is this signature, than it will be not clear. If it is some other signature, you can make it clear already now, or?

Reply:
Currently, when a signed cert is generated (see method `Ca.generateSignedCert` for example), it fills the `CertAndKey` instance passing key and cert. I would expect to pass the generated (code to write) keystore and password. So it's the caller which knows what kind of store is in the `store` field of the `CertAndKey` instance. I see your point but we cannot call it keystore or truststore because it depends. We could have a `type` field? Or we could provide two distinct fields for different stores? Is this something you are suggesting?

Reply:
I think you can have for example both set of fields and set one of them to null depending of what do you need.

> So it's the caller which knows what kind of store is in the store field of the CertAndKey instance.

The caller method might know it. But the person reading the code will be probably confused, or?

Reply:
I agree so it was my second suggestion. I guess that just one password field is enough. I don't see a scenario where we have a `CertAndKey` instance with both keystore and truststore and maybe the need for different passwords.

Reply:
Agree that different passwords are probably not needed.

Reply:
Ok I did last change, I hope we are good to go now :-)